modules/components/analytics/core/org.wso2.carbon.pc.analytics.core/src/main/java/org/wso2/carbon/pc/analytics/core/ProcessLevelMonitor.java

```diff
@@ -0,0 +1,501 @@
+				}
+			}
+		} catch (Exception e) {
+			String errMsg = "PC Analytics core ProcessLevelMonitoring error.";
+			log.error(errMsg, e);
+		}
+		if (log.isDebugEnabled()) {
+			log.debug("Result = " + sortedResult);
+		}
+		return sortedResult;
+	}
+
+	/**
+	 * perform query: SELECT processDefinitionId, COUNT(processInstanceId) AS processInstanceCount
+	 * FROM PROCESS_USAGE_SUMMARY WHERE <date range> GROUP BY processDefinitionId;
+	 *
+	 * @param filters is a given date range represents as the JSON string
+	 * @return the result as a JSON string
+	 */
+	public String getProcessInstanceCountVsProcessId(String filters) {
+		String sortedResult = "";
+		try {
+			if (Helper.isDASAnalyticsActivated()) {
+				JSONObject filterObj = new JSONObject(filters);
+				long from = filterObj.getLong(AnalyticConstants.START_TIME);
+				long to = filterObj.getLong(AnalyticConstants.END_TIME);
+				String order = filterObj.getString(AnalyticConstants.ORDER);
+				int processCount = filterObj.getInt(AnalyticConstants.NUM_COUNT);
+
+				AggregateField countField = new AggregateField();
+				countField.setFieldName(AnalyticConstants.ALL);
+				countField.setAggregate(AnalyticConstants.COUNT);
+				countField.setAlias(AnalyticConstants.PROCESS_INSTANCE_COUNT);
+
+				ArrayList<AggregateField> aggregateFields = new ArrayList<>();
+				aggregateFields.add(countField);
+
+				AggregateQuery query = new AggregateQuery();
+				query.setTableName(AnalyticConstants.PROCESS_USAGE_TABLE);
+				query.setGroupByField(AnalyticConstants.PROCESS_DEFINITION_KEY);
+				if (from != 0 && to != 0) {
+					query.setQuery(Helper.getDateRangeQuery(AnalyticConstants.COLUMN_FINISHED_TIME,
+					                                        from, to));
+				}
+				query.setAggregateFields(aggregateFields);
+
+				if (log.isDebugEnabled()) {
+					log.debug(Helper.getJSONString(query));
+				}
+
+				String result = AnalyticsRestClient
+								.post(Helper.getURL(AnalyticConstants.ANALYTICS_AGGREGATE),
+						        Helper.getJSONString(query));
+
+				JSONArray unsortedResultArray = new JSONArray(result);
+				Hashtable<String, Integer> table = new Hashtable<>();
+
+				if (unsortedResultArray.length() != 0) {
+					for (int i = 0; i < unsortedResultArray.length(); i++) {
+						JSONObject jsonObj = unsortedResultArray.getJSONObject(i);
+						JSONObject values = jsonObj.getJSONObject("values");
+						String processDefKey = values.getJSONArray("processDefKey").getString(0);
+						int processInstanceCount = values.getInt("processInstanceCount");
+						table.put(processDefKey, processInstanceCount);
+					}
+					sortedResult = Helper.getIntegerValueSortedList(table, "processDefKey",
+					                                                "processInstanceCount", order,
+					                                                processCount);
+				}
+			}
+		} catch (Exception e) {
+			String errMsg = "PC Analytics core ProcessLevelMonitoring error.";
```

Comment:
Do we need this errMsg variable? 
can we use log.error("PC Analytics core ProcessLevelMonitoring error.", e);